src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -538,8 +540,119 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (this.inErrorState) {
 			return undefined;
 		}
+
 		let cell = this.createCell(cellType);
+		return this.insertCell(cell, index);
+	}
+
+	public splitCell(cellType: CellType, index?: number, notebookService?: INotebookService): ICellModel | undefined {
```

Comment:
Please add tests in notebookModel.test.ts

Reply:
This is taking some time. I want to check this in so that users can test it out.
 I've created a separate task to add test https://github.com/microsoft/azuredatastudio/issues/17267